hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -391,6 +395,18 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 	reportSilecedAlerts(hcluster)
 	reportProxyConfig(hcluster)
 
+	// Add a grace-period finalizer
+	if hcluster.Annotations == nil {
+		hcluster.Annotations = make(map[string]string)
+	}
+
+	if gracePeriod, ok := hcluster.Annotations[hyperv1.HCDestroyGracePeriodAnnotation]; ok && len(gracePeriod) > 0 && hcDestroyGracePeriod == 0 {
+		hcDestroyGracePeriod, err = time.ParseDuration(gracePeriod)
+		if err != nil {
```

Comment:
Why is this here vs within the deletion timestamp check? do we want to parse and block if it fails here?

Reply:
Because I think it's a good UX to have the capability to modify the gracePeriod when the user needs. About the parse and block, I think it's worth for the user to know if the annotation is well set, but I don't have preference there.